File: data-structures-py/graphs/DepthFirstSearch/depth_first_search.py


class Node:
  def __init__(self, name):
    self.name = name
    self.children = []
  
  def add_child(self,name):
    self.children.append(Node(name))
    
  def depth_first_search(self,array):
    array.append(self.name)
    for child in self.children:
      child.depth_first_search(array)
    return array

File: data-structures-js/Graphs/DepthFirstSearch/DepthFirstSearch.js



class Node {
  constructor(name) {
    this.name = name;
    this.children = [];
  }

  addChild(name) {
    this.children.push(new Node(name));
    return this;
  }

  depthFirstSearch(array) {
    // Write your code here.
    // each node is a vertex - each line or connection is an edge
    // Time = O(v + e)
    // Space = O(v)
		array.push(this.name)
		for (const child of this.children){
			child.depthFirstSearch(array)
		}
		return array
  }
}